### PR TITLE
[fix])(function) add function regexp_extract_or_null

### PR DESCRIPTION
**Pull request by suxiaogang223**
## Proposed changes

Add function regexp_extract_or_null to be compatible with presto. The function is same as regexp_extract, except that it returns null when no match is found.

### CODE REVIEW

**Comment by suxiaogang223** [comment]
run buildall

**Review by wuwenchi — APPROVED** [review]

File: be/src/vec/functions/function_regexp.cpp [hunk 1]
```diff
@@ -277,6 +277,101 @@ struct RegexpExtractImpl {
     }
 };
 
+// To be compatible with presto.
+// The function is same as RegexpExtract, except that it returns null when no match is found.
```
Comment (HappenLee) [inline be/src/vec/functions/function_regexp.cpp:281]: no need copy the code。you should use template <bool return_null>to do the work
```
static constexpr auto name = return_null ? "regexp_extract_or_null" : "regexp_extract"
```

**Comment by suxiaogang223** [comment]
run buildall

**Comment by morrySnow** [comment]
add function doc to [apache/doris-website](https://github.com/apache/doris-website)

**Review by morningman — APPROVED** [review]
LGTM

**Comment by suxiaogang223** [comment]
> add function doc to [apache/doris-website](https://github.com/apache/doris-website)

https://github.com/apache/doris-website/pull/1005

**Pull request merged**